src/_P061_KeyPad.ino

```diff
@@ -97,28 +97,22 @@ boolean Plugin_061(uint8_t function, struct EventStruct *event, String& string)
       break;
     }
 
+    case PLUGIN_I2C_HAS_ADDRESS:
     case PLUGIN_WEBFORM_SHOW_I2C_PARAMS:
     {
-      uint8_t addr = PCONFIG(0);
+      const int i2cAddressValues[] = { 0x20, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x38, 0x39, 0x3A, 0x3B, 0x3C, 0x3D, 0x3E, 0x3F };
+      if (function == PLUGIN_WEBFORM_SHOW_I2C_PARAMS) {
+        addFormSelectorI2C(F("i2c_addr"), (PCONFIG(1) == 0) ? 8 : 16, i2cAddressValues, PCONFIG(0));
 
-      int optionValues[16] = { 0x20, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x38, 0x39, 0x3A, 0x3B, 0x3C, 0x3D, 0x3E, 0x3F };
-      addFormSelectorI2C(F("i2c_addr"), (PCONFIG(1) == 0) ? 8 : 16, optionValues, addr);
-
-      if (PCONFIG(1) != 0) {
-        addFormNote(F("PCF8574 uses address 0x20+; PCF8574<b>A</b> uses address 0x38+"));
+        if (PCONFIG(1) != 0) {
+          addFormNote(F("PCF8574 uses address 0x20+; PCF8574<b>A</b> uses address 0x38+"));

```

Comment:
I don't understand this note.
It suggests you need to use the last 8 of the addresses, but you're showing the first 8.

Reply:
That's an existing note... I assumed it to be correct 🤔

Reply:
The note is not wrong, though maybe a bit unclear:
- PCF8574 supports addresses from `0x20..0x27`
- PCF8574A supports addresses from `0x38..0x3F`

The scanner should report all, as this plugin supports both chips, the only difference is the supported address range.